### PR TITLE
Add Intellij files that shouldn't have been ignored

### PR DESCRIPTION
**Pull request by matozoid**
See https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems

This is so we can mark directories as ignored etc. and share that setting.

### CODE REVIEW

**Comment by ftomassetti** [comment]
Ok, it makes sense. Normally I regenerate project configurations from Maven or Gradle so I typically do not include them but this makes sense and I read the explanation you shared

File: .gitignore [hunk 1]
```diff
@@ -6,24 +6,39 @@ target
...
 release.properties
 **/.DS_Store
-*.class
-
-# Mobile Tools for Java (J2ME)
-.mtj.tmp/
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
-target
```
Comment (ftomassetti) [inline .gitignore:20]: is the target directory generated by maven? Could we want to exclude it for some reason?

File: .idea/codeStyleSettings.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ftomassetti) [inline .idea/codeStyleSettings.xml:1]: Nice!

**Review by ftomassetti — APPROVED** [review]
Sounds good

File: .gitignore (hunk 1 above)
Comment (matozoid) [reply .gitignore:20]: It was duplicated, so this is just deduplicating it. Target is still ignored :-)

File: .idea/codeStyleSettings.xml (hunk 2 above)
Comment (matozoid) [reply .idea/codeStyleSettings.xml:1]: What? XML?

**Pull request merged**